pkg/networks/usernet/client.go

```diff
@@ -74,11 +76,18 @@ func (c *Client) ResolveAndForwardSSH(ipAddr string, sshPort int) error {
 }
 
 func (c *Client) ResolveIPAddress(ctx context.Context, vmMacAddr string) (string, error) {
-	timeout := time.After(2 * time.Minute)
+	timeout := 2 * time.Minute
+	customTimeout := os.Getenv("CUSTOM_TIMEOUT")
+	if customTimeout != "" {
+		if parsedTimeout, err := strconv.Atoi(customTimeout); err == nil {
+			timeout = time.Duration(parsedTimeout) * time.Minute
+		}
+	}
```

Comment:
We can introduce the field `ResolveIPAddressTimeout` with the type `time.Duration` in the `Client` struct.

Reply:
Initialization will be done in this function again?